File: packages/luciq_flutter/ios/Classes/Modules/FeatureRequestsApi.m
#import "LuciqSDK/LuciqSDK.h"
#import "FeatureRequestsApi.h"
#import "ArgsRegistry.h"
#import "../Util/LuciqFlutterLogger.h"
#import "../Util/LuciqFlutterDebugTags.h"

extern void InitFeatureRequestsApi(id<FlutterBinaryMessenger> messenger) {
    FeatureRequestsApi *api = [[FeatureRequestsApi alloc] init];
    FeatureRequestsHostApiSetup(messenger, api);
}

@implementation FeatureRequestsApi

- (void)showWithError:(FlutterError *_Nullable *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags featureRequests] format:@"[FR.show] phase=enter"];
    [LCQFeatureRequests show];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags featureRequests] format:@"[FR.show] phase=exit"];
}

- (void)setEmailFieldRequiredIsRequired:(NSNumber *)isRequired actionTypes:(NSArray<NSString *> *)actionTypes error:(FlutterError *_Nullable *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags featureRequests] format:@"[FR.setEmailFieldRequired] phase=enter isRequired=%@ actionTypesCount=%lu", ([isRequired boolValue] ? @"true" : @"false"), (unsigned long)actionTypes.count];
    LCQAction resolvedTypes = 0;

    for (NSString *type in actionTypes) {
        NSNumber *mapped = ArgsRegistry.actionTypes[type];
        if (mapped == nil) {
            [LuciqFlutterLogger w:[LuciqFlutterDebugTags featureRequests] format:@"[FR.setEmailFieldRequired] phase=warn errorType=UnknownEnum actionType=%@", type];
            continue;
        }
        resolvedTypes |= mapped.integerValue;
    }

    [LCQFeatureRequests setEmailFieldRequired:[isRequired boolValue] forAction:resolvedTypes];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags featureRequests] format:@"[FR.setEmailFieldRequired] phase=exit"];
}

@end

File: packages/luciq_flutter/android/src/main/java/ai/luciq/flutter/modules/FeatureRequestsApi.java
package ai.luciq.flutter.modules;

import android.annotation.SuppressLint;

import androidx.annotation.NonNull;

import ai.luciq.featuresrequest.FeatureRequests;
import ai.luciq.flutter.generated.FeatureRequestsPigeon;
import ai.luciq.flutter.util.ArgsRegistry;
import ai.luciq.flutter.util.LuciqFlutterDebugTags;
import ai.luciq.flutter.util.LuciqFlutterLogger;

import java.util.List;

import io.flutter.plugin.common.BinaryMessenger;

public class FeatureRequestsApi implements FeatureRequestsPigeon.FeatureRequestsHostApi {

    public static void init(BinaryMessenger messenger) {
        final FeatureRequestsApi api = new FeatureRequestsApi();
        FeatureRequestsPigeon.FeatureRequestsHostApi.setup(messenger, api);
    }

    @Override
    public void show() {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.FEATURE_REQUESTS,
                "[FR.show] phase=enter");
        FeatureRequests.show();
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.FEATURE_REQUESTS,
                "[FR.show] phase=exit");
    }

    @SuppressLint("WrongConstant")
    @Override
    public void setEmailFieldRequired(@NonNull Boolean isRequired, @NonNull List<String> actionTypes) {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.FEATURE_REQUESTS,
                "[FR.setEmailFieldRequired] phase=enter isRequired=" + isRequired
                        + " actionTypesCount=" + actionTypes.size());
        int[] actions = new int[actionTypes.size()];
        for (int i = 0; i < actionTypes.size(); i++) {
            Integer mapped = ArgsRegistry.actionTypes.get(actionTypes.get(i));
            if (mapped == null) {
                LuciqFlutterLogger.w(LuciqFlutterDebugTags.FEATURE_REQUESTS,
                        "[FR.setEmailFieldRequired] phase=warn errorType=UnknownEnum actionType="
                                + actionTypes.get(i));
                continue;
            }
            actions[i] = mapped;
        }

        FeatureRequests.setEmailFieldRequired(isRequired, actions);
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.FEATURE_REQUESTS,
                "[FR.setEmailFieldRequired] phase=exit");
    }
}
